### ISSUE TITLE
Preserve request chaining when duplicating or importing workspaces

### ISSUE THREAD

**Issue by ryancocuzzo** [getinsomnia/insomnia#5730]
### Expected Behavior

Let's say you have a folder A of chained requests.
```
/A
  - r1
  - r2 (depends on r1 response)
```
When you duplicate that folder (creating new folder `B`), the chained links should update properly:
```
/A
  - r1
  - r2 (depends on r1 response)
/B
  - r1
  - r2 (depends on B/r1 response)
```

### Actual Behavior

```
/A
  - r1
  - r2 (depends on r1 response)
```
When you duplicate that folder (creating new folder `B`), the chained links don't update properly:
```
/A
  - r1
  - r2 (depends on r1 response)
/B
  - r1
  - r2 (depends on A/r1 response) <---- should depend on B/r1 response)
```

### Reproduction Steps

Create a folder with 2 requests.
The second request should have a variable that depends on the first response body.
Duplicate it.

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

2022.7.5

### What operating system are you using?

macOS

### Operating System Version

12.6

### Installation method

Download from Insomnia site

**Comment by PedroLuisBrilhadori** [comment]
This problem also affects environmental variables

```
/A
  - r1
  - r2 (depends on r1 response)
  - env (depends on r2 response id)
/B
  - r1
  - r2 (depends on A/r1 response) <---- should depend on B/r1 response)
  - env (depends on A/r2 response id)
```

The ids are globally unique, would it be possible to change this for each document?

In document A
r1 has the id x

In document B r1 has the id y.

A possible solution would be for the requests and properties of the document to be unique only within it.

With this we can have more than one request with the same ID, but not in the same document

**Comment by rezamanshouri** [comment]
Any updates here?

Issue `5942` referenced above is still a pain when needing to share collections (export/import).

**Comment by Stratum-Jeremy** [comment]
I have the same issue but noticed something strange.

If I export a collection and import it as a new collection on the same Insomnia instance I exported it, the reference IDs are maintained. However when I send the export file to a coworker and they import the collection the reference IDs are reassigned and the tags break.

**Comment by PedroLuisBrilhadori** [comment]
@Stratum-Jeremy 

When you export a collection that you export yourself, the IDs exist in the local database and therefore work perfectly.

Test: 
 1. export collection,
 2. delete old collection,
 3. import. 
 This way the error will occur

**Comment by efeevren** [comment]
This bug is blocking to work with team, every member tries to fix same problem sometimes success sometimes failure. We choose to change tool.

**Comment by bwaidelich** [comment]
This issue has been reported multiple times and there is a [PR](https://github.com/Kong/insomnia/pull/6521) for it. Can someone give an update what is required for this to proceed?

**Comment by Yocker95k** [comment]
Yes, this is a big problem for collaborative work. We will change the tool if this is not fixed soon.

**Comment by UnicodeTreason** [comment]
Started trying to share Collections with team mates today and hit this issue, is there an ETA on resolution?
Really kills the usefulness of the app if a junior can't import an API Collection to see how an API works.

**Comment by tl-luca-mazzanti** [comment]
Any chance to have this thing fixed? It's impossible to share collections

**Comment by AnnaGolosova** [comment]
Any updates here? It's impossible to share collections

**Comment by BlaineCharter** [comment]
Switching software, this won't work.

**Comment by karivatj** [comment]
Any updates on this? Request chains still break after exporting and importing in the most recent version (v9.3.3), making team collaboration impossible.

It appears that the referenced request IDs remain intact after export, but new request IDs are generated for some reason, and the references are not updated, which breaks the entire collection of chained requests.

Can someone seriously address this? Insomnia is a great tool, but this fundamentally destroys team collaboration since we cannot share collections properly.

**Comment by PedroLuisBrilhadori** [comment]
a temporary solution is to build the version of this PR #6521. Doesn't completely fix it, but it works

**Comment by Diegovictorbr** [comment]
Any ETA for this one? Sharing Insomnia collections consistently across teams is impossible.

**Comment by tauinger-de** [comment]
Is this issue intentionally left open to make users switch to paid plans? It's nearly two years since this ticket was opened

**Comment by PedroLuisBrilhadori** [comment]
this is definitely not a priority for them. 

But no thought was given to exactly how to correct it.

The software would have to make a change that requires a lot of effort to resolve. I believe that what we can do is try to think and better document a possible solution for this case to move forward.

**Comment by timhansinger** [comment]
Yes, it's really bad to have such a major flaw in the software.
I'm currently trying out Bruno (haven't had much time yet to compare the feature set though) as an alternative since this is such a painful bug.

**Comment by ivanlonel** [comment]
Shh... We don't talk about Bruno.

**Comment by Brazlucas** [comment]
Nothing yet.. How is it possible...

**Comment by AnnaGolosova** [comment]
Any updates here? It's impossible to share collections

**Comment by Hotbarango** [comment]
Another option for a quick fix to make things useable again is to just add a feature to mass update the ids.

When a chained request detects that it's pointing to a request id that no longer exists
- Allow the option for the user to indicate the correct request
- mass update all others in the same collection that were pointing to that deleted request as well.

**Comment by ben12** [comment]
I finally choose to abandon insomnia for testing 🥹, too many issues with insomnia and inso cli.

For testing, I am switching to [openapi generator](https://openapi-generator.tech) + Mocha + Chai. Free, easy, fast, flexible and collaborative.

**Comment by bjarnesvanberg** [comment]
I have been trying to convince my company to use this software and Kong's platform for a long time, but _"stuff"_ like this makes it impossible to convince anyone 🤮 

This just makes me a firm believer that _any_ alternative would be better in the long run. I have heard Bruno is coming along pretty well.

**Comment by seroperson** [comment]
@gatzjames @jackkav can I give a try on resolving this? The previous PR was closed and you said it'll be fixed in another PR, that's why I'm asking, maybe you have plans on this.
